core/src/main/scala/cromwell/core/io/AsyncIo.scala

```diff
@@ -0,0 +1,78 @@
+package cromwell.core.io
+
+import akka.actor.{Actor, ActorLogging, ActorRef}
+import cromwell.core.path.BetterFileMethods.OpenOptions
+import cromwell.core.path.Path
+
+import scala.concurrent.duration._
+import scala.concurrent.{Future, Promise}
+import scala.language.postfixOps
+
+
+trait AsyncIo extends IoClientHelper { this: Actor with ActorLogging with IoCommandBuilder =>
+  
+  protected val ioTimeout = 3 minutes
+  
+  override private [core] def ioResponseReceive: Receive = {
+    case (promise: Promise[_], ack: IoAck[Any] @unchecked) =>
+      cancelTimeout(promise -> ack.command)
+      // This is not typesafe. 
+      // However the sendIoCommand method ensures that the command and the promise have the same generic type
+      // Which means as long as only the sendIoCommand method is used to send requests, and the ioActor honors his contract
+      // and send back the right context with the right response, the types are virtually guaranteed to match.
+      promise.asInstanceOf[Promise[Any]].complete(ack.toTry)
+      ()
+  }
+  
+  /**
+    * IMPORTANT: This loads the entire content of the file into memory !
+    * Only use for small files !
+    */
+  def contentAsStringAsync(path: Path): Future[String] = {
+    val promise = Promise[String]
+    sendIoCommandWithPromise(contentAsStringCommand(path), promise)
+    promise.future
+  }
+  
+  def writeAsync(path: Path, content: String, options: OpenOptions): Future[Unit] = {
+    val promise = Promise[Unit]
+    sendIoCommandWithPromise(writeCommand(path, content, options), promise)
+    promise.future
+  }
+
+  def sizeAsync(path: Path): Future[Long] = {
+    val promise = Promise[Long]
+    sendIoCommandWithPromise(sizeCommand(path), promise)
+    promise.future
+  }
+
+  def hashAsync(path: Path): Future[String] = {
+    val promise = Promise[String]
+    sendIoCommandWithPromise(hashCommand(path), promise)
+    promise.future
+  }
+
+  def deleteAsync(path: Path, swallowIoExceptions: Boolean = false): Future[Unit] = {
+    val promise = Promise[Unit]
+    sendIoCommandWithPromise(deleteCommand(path, swallowIoExceptions), promise)
+    promise.future
+  }
+
+  def copyAsync(src: Path, dest: Path, overwrite: Boolean = true): Future[Unit] = {
+    val promise = Promise[Unit]
+    // Allow for a much larger timeout for copies, as large files can take a while (even on gcs, if they are in different locations...)
+    sendIoCommandWithPromise(copyCommand(src, dest, overwrite), promise, 1 hour)
```

Comment:
I could see both. I think it could confuse the user to have it as a knob because it's more of an internal timeout. It's a cromwell actor timing out because another cromwell actor didn't respond in time. If I would see a `ioTimeout` knob I would think it means `how much time will cromwell wait before timing out when it's contacting an external service to perform I/O`. Which is controlled by a different variable: https://github.com/broadinstitute/cromwell/pull/2056/files#diff-5cad4efaf6fca811a2af3df60083eccfR39